### PR TITLE
feat[smock]: add a wallet that be used to send transactions from a smock

### PR DESCRIPTION
**Pull request by smartcontracts**
**Description**
Was working on this for fun over the weekend. Adds a `wallet` to every smock that can be used to send transactions *from* the smock account.

### CODE REVIEW

File: packages/smock/README.md [hunk 1]
```diff
@@ -51,40 +51,6 @@ const config: HardhatUserConfig = {
 export default config
 ```
 
-## Table of Contents
```
Comment (smartcontracts) [inline packages/smock/README.md:54]: GitHub handles ToC relatively well now so we can just get rid of this.

File: packages/smock/README.md [hunk 2]
```diff
@@ -339,6 +308,20 @@ MyMockContract.smocked.myFunction.will.revert.with('0x1234')
...
+import { ethers } from 'hardhat'
+import { smockit } from '@eth-optimism/smock'
+
+const myContractFactory = await ethers.getContractFactory('MyContract')
+const myContract = await myContractFactory.deploy(...)
+
+// Smockit!
+const mock = await smockit('MyContract')
+
+await myContract.connect(mock.wallet).doSomeFunction() // msg.sender == MyMockContract.address
```
Comment (smartcontracts) [inline packages/smock/README.md:322]:
```suggestion
await myContract.connect(mock.wallet).doSomeFunction() // msg.sender == mock.address
```

File: packages/smock/README.md [hunk 3]
```diff
@@ -339,6 +308,20 @@ MyMockContract.smocked.myFunction.will.revert.with('0x1234')
...
+### Sending transactions from a smocked contract
+```typescript
+import { ethers } from 'hardhat'
+import { smockit } from '@eth-optimism/smock'
+
+const myContractFactory = await ethers.getContractFactory('MyContract')
+const myContract = await myContractFactory.deploy(...)
+
+// Smockit!
+const mock = await smockit('MyContract')
```
Comment (maurelian) [inline packages/smock/README.md:320]:
```suggestion
const mock = await smockit('AnotherContract')
```

The current example makes it seem like there's some relationship between mock and myContract, which I don't think is the case.

File: packages/smock/README.md (hunk 1 above)
Comment (maurelian) [reply packages/smock/README.md:54]: It does? How/where?

File: packages/smock/test/smockit/sending-from-mocks.spec.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  it('should attach a signer for a mock with a fixed address', async () => {
+    const mock = await smockit('TestHelpers_BasicReturnContract', {
+      address: '0x1234123412341234123412341234123412341234'
+    })
+
+    expect(
+      await TestHelpers_SenderAssertions.connect(mock.wallet).getSender()
+    ).to.equal(mock.address)
+  })
+})
```
Comment (maurelian) [inline packages/smock/test/smockit/sending-from-mocks.spec.ts:38]: this is super useful!

**Review by maurelian — APPROVED** [review]
LGTM

File: packages/smock/README.md (hunk 1 above)
Comment (smartcontracts) [reply packages/smock/README.md:54]: Screen Shot 2021-05-18 at 10 54 12 AM

Comment (maurelian) [reply packages/smock/README.md:54]: oh nice!

**Pull request merged**